### PR TITLE
Refactored job store string to locator (resolves #864, resolves #991, resolves #992)

### LINKED ISSUE #864: Job store sytax not documented
@cmarkello wanted to use the Azure job store, but the fact that to use it you need to put `azure:<account>:<container prefix>` doesn't seem to be in the documentation. I didn't find anything about the syntax for AWS either.

The syntax to invoke the different job stores, and the meaning of their configuration strings, should be prominently documented.

### LINKED ISSUE #991: Job store string should be refactored to job store locator

### LINKED ISSUE #992: Five option help messages note incorrect exception
A few option help messages note ```ExistingJobStoreException``` as an exception that is raised if a job store is started when it already exists. The correct exception is ```JobStoreCreationException```.

### PR DESCRIPTION
**Pull request by JakeNarkizian**
Job store string has now been refactored completely. The syntax for each job store locator has been documented in the job store option help message. Minor corrections to outdated documentation have also been made.

### CODE REVIEW

**Comment by JakeNarkizian** [comment]
Error tracked in #760

**Comment by JakeNarkizian** [comment]
@hannes-ucsc ready

**Comment by JakeNarkizian** [comment]
@hannes-ucsc since the property methods were axed the documentation only shows up in the job store help method. Is this enough?

File: src/toil/common.py [hunk 1]
```diff
@@ -224,14 +224,19 @@ def _addOptions(addGroupFn, config):
...
     addOptionFn('jobStore', type=str,
-                      help=("Store in which to place job management files \
-                      and the global accessed temporary files"
-                      "(If this is a file path this needs to be globally accessible "
-                      "by all machines running jobs).\n"
-                      "If the store already exists and restart is false an"
-                      " ExistingJobStoreException exception will be thrown."))
+                help=("Store in which to place job management files and the global accessed "
+                      "temporary files. Job store locator strings should be formatted as follows\n"
+                      "aws:<aws region>:<name prefix>\n"
```
Comment (hannes-ucsc) [inline src/toil/common.py]: AWS

Comment (hannes-ucsc) [inline src/toil/common.py]: SOrry, `AWS region` instead of `aws region`

**Comment by hannes-ucsc** [comment]
> only shows up in the job store help method. Is this enough?

Yes.

**Comment by JakeNarkizian** [comment]
@hannes-ucsc ready

File: src/toil/test/jobStores/jobStoreTest.py [hunk 2]
```diff
@@ -741,7 +741,7 @@ class GoogleJobStoreTest(AbstractJobStoreTest.Test):
 
     def _createJobStore(self, config=None):
         from toil.jobStores.googleJobStore import GoogleJobStore
-        return GoogleJobStore.createJobStore(self.namePrefix + ":" + GoogleJobStoreTest.projectID,
+        return GoogleJobStore.createJobStore(GoogleJobStoreTest.projectID + ":" + self.namePrefix,
```
Comment (hannes-ucsc) [inline src/toil/test/jobStores/jobStoreTest.py]: Why did you swap these two?

Comment (hannes-ucsc) [inline src/toil/test/jobStores/jobStoreTest.py]: Never mind. Makes sense.

**Pull request merged**